txscript/data/script_tests.json

```diff
@@ -551,6 +556,370 @@
+["1", "FROMALTSTACK", "NONE", "ERR_INVALID_STACK_OPERATION"],
+
+["2DROP test coverge"],
+["0 0", "2DROP 1", "NONE", "OK"],
+["OP_2 OP_4", "2DROP 1", "NONE", "OK"],
+["DATA_15 0x7a{15}  DATA_7 0x7a{7}", "2DROP 1", "NONE", "OK"],
+["1", "2DROP 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["DATA_7 0x7a{7}", "2DROP 1", "NONE", "ERR_INVALID_STACK_OPERATION", "2DROP requires two inputs"],
+["", "2DROP 1", "NONE", "ERR_INVALID_STACK_OPERATION", "2DROP requires two inputs"],
+
+["2DUP test coverge"],
+["0 1", "2DUP", "NONE", "OK"],
+["OP_2 OP_4", "2DUP 2DROP ADD 6 EQUAL", "NONE", "OK"],
+["OP_2 OP_2", "2DUP ADD ADD ADD 8 EQUAL", "NONE", "OK"],
+["13 14", "2DUP ROT EQUALVERIFY EQUAL", "NONE", "OK"],
+["NOP", "2DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["1", "2DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["1", "2DUP", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["DATA_7 0x7a{7}", "2DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION", "2DUP requires two inputs"],
+["", "2DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION", "2DUP requires two inputs"],
+["PUSHDATA4 0x01400000 0x62{16385}", "2DUP 1", "NONE", "ERR_SCRIPT_SIZE", "Exceed max allowed script size"],
+["PUSHDATA4 0x01200000 0x62{8193}", "2DUP 1", "NONE", "ERR_PUSH_SIZE", "Exceed max allowed input size"],
+
+["3DUP test coverge"],
+["0 0 1", "3DUP", "NONE", "OK"],
+["OP_1 OP_1 OP_1", "3DUP ADD ADD ADD ADD ADD 6 EQUAL", "NONE", "OK"],
+["-1 0 1 2", "3DUP DEPTH 7 EQUALVERIFY ADD ADD 3 EQUALVERIFY 2DROP 0 EQUALVERIFY", "NONE", "OK"],
+["NOP", "3DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["1", "3DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["1 2", "3DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["1 1", "3DUP", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["DATA_7 0x7a{7} DATA_15 0x7a{15}", "3DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION", "3DUP requires three inputs"],
+["", "3DUP 1", "NONE", "ERR_INVALID_STACK_OPERATION", "3DUP requires three inputs"],
+["1 2 3 4 5 <3DUP>{173}", "1 2 3 4 5 6 <3DUP>{165}", "NONE", "ERR_STACK_SIZE", ">1,024 stack size"],
+["1 2 3 4 5 <3DUP>{172}", "1 TOALTSTACK 2 TOALTSTACK 3 4 5 6 <3DUP>{166}", "NONE", "ERR_STACK_SIZE", ">1,024 stack+altstack size"],
+["PUSHDATA4 0x01400000 0x62{16385}", "3DUP 1", "NONE", "ERR_SCRIPT_SIZE", "Exceed max allowed script size"],
+["PUSHDATA4 0x01200000 0x62{8193}", "3DUP 1", "NONE", "ERR_PUSH_SIZE", "Exceed max allowed input size"],
+
+["2OVER test coverge"],
+["0 1 0 0", "2OVER", "NONE", "OK"],
+["1 2 3 5", "2OVER ADD ADD 8 EQUALVERIFY ADD ADD 6 EQUAL", "NONE", "OK"],
+["NOP", "2OVER 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["1", "2 3 2OVER 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["1 1 1", "2OVER", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["0 1 0", "2OVER 1", "NONE", "ERR_INVALID_STACK_OPERATION", "2OVER requires two inputs"],
+["", "2OVER 1", "NONE", "ERR_INVALID_STACK_OPERATION", "2OVER requires two inputs"],
```

Comment:
Same as above with comment being wrong.

Also, I would suggest reording these so that the number of arguments supplied increase in order.

e.g. 0 args, 1 arg (needs a test), 2 args (needs a test), then the 3 arg tests.